upstream-seed/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+use anyhow::Context;
+
+use librad::profile::Profile;
+use link_identities::git::Urn;
+
+mod cli;
+mod peer;
+
+#[tokio::main(flavor = "multi_thread")]
+pub async fn main() {
+    init_logging();
+
+    let args: crate::cli::Args = structopt::StructOpt::from_args();
+
+    if let Err(err) = run(args).await {
+        tracing::error!(?err, "fatal error");
+        std::process::exit(1);
+    }
+
+    std::process::exit(0);
+}
+
+pub async fn run(options: cli::Args) -> anyhow::Result<()> {
+    let git_version = std::process::Command::new("git")
+        .arg("version")
+        .output()
+        .context("failed to run git")?
+        .stdout;
+    tracing::info!(git_version = std::str::from_utf8(&git_version).unwrap().trim());
+
+    let profile = Profile::from_root(&options.rad_home, None)
+        .context("failed to initialize Radicle profile")?;
+    let rad_paths = profile.paths().clone();
+
+    let key_path = if let Some(key_path) = options.identity_key {
+        key_path
+    } else {
+        rad_paths.keys_dir().join("identity.key")
+    };
+
+    let key = load_or_create_secret_key(&key_path)?;
+
+    let peer_id = librad::PeerId::from(&key);
+    tracing::info!(?peer_id);
+
+    tracing::info!(bootstrap = ?options.bootstrap);
+
+    let peer = peer::Peer::new(peer::Config {
+        rad_paths,
+        key,
+        listen: options.listen,
+    });
+
+    let track_task = tokio::spawn(track_projects(peer.clone(), options.project));
+
+    let bootstrap_addrs = options.bootstrap.clone().unwrap_or_default();
```

Comment:
What would this default to if the unwrap fails?

Reply:
The empty vector.

Reply:
Gotcha.